src/include/storage/data_table.h

```diff
@@ -246,7 +256,10 @@ class DataTable {
   void AtomicallyWriteVersionPtr(TupleSlot slot, const TupleAccessStrategy &accessor, UndoRecord *desired);
 
   // Checks for Snapshot Isolation conflicts, used by Update
-  bool HasConflict(UndoRecord *version_ptr, const transaction::TransactionContext *txn) const;
+  bool HasConflict(const transaction::TransactionContext &txn, UndoRecord *version_ptr) const;
+
+  // Wrapper around the other HasConflict for indexes to call (they only have tuple slot, not the version pointer)
+  bool HasConflict(const transaction::TransactionContext &txn, TupleSlot slot) const;
```

Comment:
Do we ever call `IsVisible` or `HasConflict` without calling the other?  I'm wondering if the indexes only need a single function that answers both of these questions at the same time.  Would solve question posed below.

Reply:
Reads (scans) only care about IsVisible. Writes (InsertUnique, Delete) care about IsVisible and HasConflict.